### ISSUE TITLE
Allow SMTP sending email without AUTH

### ISSUE THREAD

**Issue by MrGoodbits** [pocketbase/pocketbase#847]
First, thank you for your efforts on PocketBase, I am finding it very interesting.

I have the need to have PocketBase send emails via internal mail relays that do not require/allow authentication.  I found that whether TLS or StartTLS is used, PocketBase always attempts to "AUTH PLAIN" even when the SMTP server does not use it and when no username/password was provided.  I made the following changes to your source to only attempt to authenticate when a username and password is actually provided.

With these changes, if one doesn't provide a username and password, mailyak uses "nil" instead of a configured smtpAuth and thus it doesn't attempt to authenticate with "AUTH PLAIN" to the SMTP server.

This is working for me.

Thank you again for this project,
Mr Goodbits

# Diff for making SMTP AUTH optional
```diff
diff --git a/tools/mailer/smtp.go b/tools/mailer/smtp.go
index f21ff57..b1ec7f9 100644
--- a/tools/mailer/smtp.go
+++ b/tools/mailer/smtp.go
@@ -46,7 +46,11 @@ func (m *SmtpClient) Send(
        htmlContent string,
        attachments map[string]io.Reader,
 ) error {
-       smtpAuth := smtp.PlainAuth("", m.username, m.password, m.host)
+       smtpAuth := smtp.Auth(nil)
+       // If username and password provided, use them for authentication
+       if ((m.username != "") && (m.password != "")) {
+               smtpAuth = smtp.PlainAuth("", m.username, m.password, m.host)
+       }

        // create mail instance
        var yak *mailyak.MailYak
```

```diff
diff --git a/ui/src/components/settings/PageMail.svelte b/ui/src/components/settings/PageMail.svelte
index dd01410..6a7dd6a 100644
--- a/ui/src/components/settings/PageMail.svelte
+++ b/ui/src/components/settings/PageMail.svelte
@@ -197,6 +197,9 @@
                                 />
                             </Field>
                         </div>
+                        <div class="content txt-xl">
+                            <p>Authentication (if required)</p>
+                        </div>
                         <div class="col-lg-6">
                             <Field class="form-field" name="smtp.username" let:uniqueId>
                                 <label for={uniqueId}>Username</label>
```

**Comment by MrGoodbits** [comment]
Details:
This was a diff from 0.7.9 and tested on an Ubuntu 22.04 system.
A did a svelte npm build from ui/ for the front end and go build from examples/base/ to generate the executable to test.

**Comment by ganigeorgiev** [comment]
@MrGoodbits Thank you. I've added it in the local v0.8 branch (_without the UI changes since the fields already are optional and there is no need to clutter the interface with additional info text; if users find it confusing we can always add it later_).

The fix will be available with the v0.8.0 pre-release (sometime this weekend).
